contracts/extensions/votingReputation/VotingReputationStorage.sol

```diff
@@ -0,0 +1,342 @@
+    uint256 requiredStake = getRequiredStake(_motionId);
+
+    // Check for valid motion Id / motion
+    if (_motionId == 0 || _motionId > motionCount || motion.action.length == 0) {
+
+      return MotionState.Null;
+
+    // If finalized, we're done
+    } else if (motion.finalized) {
+
+      return MotionState.Finalized;
+
+    // Not fully staked
+    } else if (
+      motion.stakes[YAY] < requiredStake ||
+      motion.stakes[NAY] < requiredStake
+    ) {
+
+      // Are we still staking?
+      if (block.timestamp < motion.events[STAKE_END]) {
+        return MotionState.Staking;
+      // If not, did the YAY side stake?
+      } else if (motion.stakes[YAY] == requiredStake) {
+        return finalizableOrFinalized(motion.action);
+      // If not, was there a prior vote we can fall back on?
+      } else if (motion.votes[NAY] + motion.votes[YAY] > 0) {
+        return finalizableOrFinalized(motion.action);
+      // Otherwise, the motion failed
+      } else {
+        return MotionState.Failed;
+      }
+
+    // Fully staked, go to a vote
+    } else {
+
+      if (block.timestamp < motion.events[SUBMIT_END]) {
+        return MotionState.Submit;
+      } else if (block.timestamp < motion.events[REVEAL_END]) {
+        return MotionState.Reveal;
+      } else if (
+        block.timestamp < motion.events[REVEAL_END] + escalationPeriod &&
+        motion.domainId > 1
+      ) {
+        return MotionState.Closed;
+      } else {
+        return finalizableOrFinalized(motion.action);
+      }
+    }
+  }
+
+  // If we decide that the motion is finalizable, we might actually want it to
+  //  report as finalized if it's a no-action motion.
+  function finalizableOrFinalized(bytes memory action) internal pure returns (MotionState) {
+    bytes4 sig;
+    assembly { sig := mload(add(action, 0x20)) }
+    return sig == NO_ACTION ? MotionState.Finalized : MotionState.Finalizable;
+  }
+
+  // Internal functions
+
+  function getRequiredStake(uint256 _motionId) internal view returns (uint256) {
+    return wmul(motions[_motionId].skillRep, totalStakeFraction);
+  }
+
+  function getTarget(address _target) internal view returns (address) {
+    return (_target == address(0x0)) ? address(colony) : _target;
+  }
+
+  function flip(uint256 _vote) internal pure returns (uint256) {
+    return 1 - _vote;
+  }
+
+  function isExpenditureSig(bytes4 sig) internal pure returns (bool) {
+    return sig == SET_EXPENDITURE_STATE || sig == SET_EXPENDITURE_PAYOUT;
+  }
+
+  function getSig(bytes memory action) internal pure returns (bytes4 sig) {
+    assembly {
+      sig := mload(add(action, 0x20))
+    }
+  }
+
+  function getExpenditureId(bytes memory action) internal pure returns (uint256 expenditureId) {
+    bytes4 sig = getSig(action);
+    assert(isExpenditureSig(sig));
+
+    assembly {
+      expenditureId := mload(add(action, 0x64))
+    }
+  }
+
+  function getExpenditureAction(bytes memory action) internal view returns (bytes memory) {
```

Comment:
Are you sure about the second case? The end of the `else` clause is:

```
        if (!isExpenditureSig(summary.sig)) {
          summary.sig = sig;
        }
```

Which would return the last function signature if no expenditure signatures were seen.

The comment says as much:

```
        // If no expenditure signatures have been seen, we record the latest signature
```

Also, if the action is not a multicall, then we parse the single action, returning the same summary as if it were a multicall of one action.

Reply:
Your comment would be more accurately phrased as:

- If passed an action that is a multicall, returns the first expenditure sig inside a multicall
- If passed an action that is a multicall, and there are no expenditure sigs inside, returns  the final function sig
- If passed an action that is a not multicall, returns the summary of the single action

Reply:
Ok, I see what you're saying. I think I was able to get away with it during my implementation because it's called in contexts where we either already know we're dealing with an expenditure (for example by previously checking `motion.sig`) or the result of `isExpenditureSig(getSig(action))` will be false regardless of whether the action is null or a non-expenditure. But I agree it's not an internally consistent function.

Reply:
I could add `return actions[0];` at the end of the first branch if no expenditure sigs are found, so at least the typing will be consistent. Alternatively I could add a comment explaining the intended behavior, which would be my preference.

Reply:
> rather it's not clear that this function assumes that either the action is an expenditure function or that the contents contain an expenditure function.

If this is the case, can we not `require` it?